app/governance/proposal/page.tsx

```diff
@@ -0,0 +1,315 @@
+"use client";
+
+import { useRouter, useSearchParams } from "next/navigation";
+import Text from "@/components/text";
+import styles from "./proposalModal.module.scss";
+import {
+  calculateVotePercentages,
+  formatDeposit,
+  formatProposalStatus,
+  formatProposalType,
+  formatTime,
+} from "@/utils/gov/formatData";
+import Icon from "@/components/icon/icon";
+import Button from "@/components/button/button";
+import useProposals from "@/hooks/gov/useProposals";
+import { useState } from "react";
+import { Proposal } from "@/hooks/gov/interfaces/proposal";
+import useCantoSigner from "@/hooks/helpers/useCantoSigner";
+import Splash from "@/components/splash/splash";
+import { VoteOption } from "@/hooks/gov/interfaces/voteOptions";
+import Image from "next/image";
+import { NEW_ERROR } from "@/config/interfaces";
+
+import { VotingInfoBox } from "../components/VotingInfoBox/VotingInfoBox";
+import { TransactionFlowType } from "@/transactions/flows/flowMap";
+import { NewTransactionFlow } from "@/transactions/flows/types";
+import Spacer from "@/components/layout/spacer";
+
+export default function Page() {
+  function castVote(proposalId: number, voteOption: VoteOption | null) {
+    if (signer) {
+      if (!voteOption) {
+        return NEW_ERROR("Please select a vote option");
+      }
+      //console.log("cast vote test");
+      const newFlow: NewTransactionFlow = {
+        icon: "/canto.svg",
+        txType: TransactionFlowType.VOTE_TX,
+        title: "Voting",
+        params: {
+          chainId: chainId,
+          ethAccount: signer?.account.address ?? "",
+          proposalId: proposalId,
+          voteOption: voteOption,
+        },
+      };
+      txStore?.addNewFlow({
+        txFlow: newFlow,
+        ethAccount: signer.account.address,
+      });
+    }
+  }
+
+  const searchParams = useSearchParams();
+  const id = searchParams.get("id");
+  const proposalId = Number(id);
+  const router = useRouter();
+
+  const { txStore, signer, chainId } = useCantoSigner();
+  const { proposals, isProposalsLoading } = useProposals({ chainId: chainId });
+  const [selectedVote, setSelectedVote] = useState<VoteOption | null>(null);
+
+  const [isChecked, setChecked] = useState(false);
+
+  const handleCheck = (voteOption: VoteOption) => {
+    setSelectedVote(voteOption);
+    setChecked(!isChecked);
+  };
+
+  //console.log(selectedVote);
+  if (isProposalsLoading) {
+    return <Splash />;
+  }
+
+  if (!id) {
+    return (
```

Comment:
lets just have one return. Do this all at the end

Reply:
This is actually done coz we're actually fetching the id(in query params) from Proposals Data and then if id is not found, we wouldn't need to go through all other code below as the code below wouldn't be having any data for calculations. We can make every variable in the calculations undefined in that case but I felt this would be better.